### PR TITLE
Adding a software token for testing.

### PR DESCRIPTION
**Pull request by arnar**
This adds softtokenhelper.js, an implementation of the lower half that emulates a real key with software, using e2e crypto.

### CODE REVIEW

File: u2f-chrome-extension/README.md [hunk 1]
```diff
@@ -202,3 +202,60 @@ var sign_helper_reply = {
...
+(e.g.) https://crxjs-dot-u2fdemo.appspot.com and register a new key. The log
+output will indicate the operations on the software key. You can then inspect
+the global variable `softtoken_profile` for the generated keys, counters and
+appIds.
+
+**NOTE**: The contents of `softtoken_profile` is **not persisted** across
+extension reloads. In particular all private keys will disappear on a browser
+restart.
+
+**Do not use the software token tok register keys on real accounts, as there are
```
Comment (leshi) [inline u2f-chrome-extension/README.md]: s/tok/to/

**Comment by leshi** [comment]
minor nit, but LGTM otherwise

**Comment by mschilder123** [comment]
why is DEFAULT_ATTESTATION_CERT not the minimal self-signed cert?

**Comment by arnar** [comment]
This is simply a copy of the example key that is already listed it the
specs themselves. Reason is part laziness and part not wanting to publish
another private key for verifiers to watch out for.

Why that key doesn't come with a minimal cert, I don't know.

**Pull request merged**